sdv/sampling/independent_sampler.py

```diff
@@ -0,0 +1,134 @@
+"""Independent Samplers."""
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseIndependentSampler():
+    """Independent sampler mixin.
+
+    Args:
+        metadata (sdv.metadata.multi_table.MultiTableMetadata):
+            Multi-table metadata representing the data tables that this sampler will be used for.
+        table_synthesizers (dict):
+            Dictionary mapping each table to a synthesizer. Should be instantiated and passed to
+            sampler by the multi-table synthesizer using this sampler.
+        table_sizes (dict):
+            Dictionary mapping each table to its size. Should be instantiated and passed to the
+            sampler by the multi-table synthesizer using this sampler.
+    """
+
+    def __init__(self, metadata, table_synthesizers, table_sizes):
+        self.metadata = metadata
+        self._table_synthesizers = table_synthesizers
+        self._table_sizes = table_sizes
+
+    def _add_foreign_key_columns(self, child_table, parent_table, child_name, parent_name):
+        """Add all the foreign keys that connect the child table to the parent table.
+
+        Args:
+            child_table (pd.DataFrame):
+                The table containing data sampled for the child.
+            parent_table (pd.DataFrame):
+                The table containing data sampled for the parent.
+            parent_name (str):
+                The name of the parent table.
+            child_name (str):
+                The name of the child table.
+        """
+        raise NotImplementedError()
+
+    def _sample_table(self, synthesizer, table_name, num_rows, sampled_data):
+        """Sample a single table and all its children.
+
+        Args:
+            synthesizer (SingleTableSynthesizer):
+                Synthesizer to sample from for the table.
+            table_name (string):
+                Name of the table to sample.
+            num_rows (int):
+                Number of rows to sample for the table.
+            sampled_data (dict):
+                A dictionary mapping table names to sampled tables (pd.DataFrame).
+        """
+        LOGGER.info('Sampling %s rows from table %s', num_rows, table_name)
+
+        sampled_rows = synthesizer._sample_batch(num_rows, keep_extra_columns=True)
+        sampled_data[table_name] = sampled_rows
+
+    def _connect_tables(self, sampled_data):
+        """Connect all related tables.
+
+        This method iterates over all tables and adds all foreign keys to connect
+        related tables.
+
+        Args:
+            sampled_data (dict):
+                A dictionary mapping table names to the sampled tables (pd.DataFrame).
+        """
+        queue = [
+            table for table in self.metadata.tables if not self.metadata._get_parent_map()[table]
+        ]
```

Comment:
Small nit that helps readability if you like it (not required to change up to you):
With list comprehensions you can do:
```python
queue = [
    table
    for table in self.metadata.tables
    if not self.metadata._get_parent_map()[table]
]
``` 
To me it makes it more easy to read since you have `object` that you are storing, the loop that you are performing, the condition that you apply in multiple lines.